ambry-store/src/main/java/com.github.ambry.store/DiskManager.java

```diff
@@ -0,0 +1,168 @@
+    Timer.Context context = metrics.diskStartTime.time();
+    try {
+      File mountPath = new File(disk.getMountPath());
+      if (mountPath.exists()) {
+        Map<StoreStarter, Thread> startupThreads = new HashMap<>();
+        for (final Map.Entry<PartitionId, BlobStore> partitionAndStore : stores.entrySet()) {
+          StoreStarter storeStarter = new StoreStarter(partitionAndStore.getKey(), partitionAndStore.getValue());
+          Thread thread = Utils.newThread("store-startup-" + partitionAndStore.getKey(), storeStarter, false);
+          thread.start();
+          startupThreads.put(storeStarter, thread);
+        }
+        int numFailed = 0;
+        for (Map.Entry<StoreStarter, Thread> starterAndThread : startupThreads.entrySet()) {
+          starterAndThread.getValue().join();
+          Exception exception = starterAndThread.getKey().exception;
+          if (exception != null) {
+            numFailed++;
+            logger.error("Exception while starting store for the partition" + starterAndThread.getKey().partition,
+                exception);
+          }
+        }
+        if (numFailed > 0) {
+          metrics.diskStartFailure.inc();
+          logger.error("Could not start " + numFailed + " out of " + stores.size() + " stores on the disk " + disk);
+        }
+      } else {
+        metrics.diskStartFailure.inc();
+        logger.error("Mount path does not exist: " + mountPath + " ; cannot start stores on this disk");
+      }
+    } finally {
+      context.stop();
+    }
+  }
+
+  /**
+   * Shuts down all the stores this disk.
+   */
+  void shutdown() {
+    for (Map.Entry<PartitionId, BlobStore> partitionAndStore : stores.entrySet()) {
+      try {
+        partitionAndStore.getValue().shutdown();
+      } catch (Exception e) {
+        logger.error(
+            "Exception while shutting down store for partition " + partitionAndStore.getKey() + " on disk " + disk);
+      }
+    }
+  }
+
+  /**
+   * @param id the {@link PartitionId} to find the store for.
+   * @return the associated {@link Store}, or {@code null} if the partition is not on this disk, or the store is not
+   *         started.
+   */
+  Store getStore(PartitionId id) {
+    BlobStore store = stores.get(id);
+    return store.isStarted() ? store : null;
+  }
+
+  /**
+   * @return the {@link DiskId} that is managed by this {@link DiskManager}.
+   */
+  DiskId getDisk() {
+    return disk;
+  }
+
+  /**
+   * A {@link Runnable} that starts a {@link Store} and saves a reference to an exception if starting failed.
+   */
+  private class StoreStarter implements Runnable {
+    final PartitionId partition;
+    final Store store;
+    volatile Exception exception = null;
+
+    public StoreStarter(PartitionId partition, Store store) {
+      this.partition = partition;
+      this.store = store;
+    }
+
+    @Override
+    public void run() {
+      try {
+        store.start();
+      } catch (StoreException e) {
```

Comment:
Don't you want to catch all kinds of exceptions here? Particularly since certain I/O errors do not manifest as `StoreException`s.